src/Jackett.Common/Indexers/BJShare.cs

```diff
@@ -520,7 +531,17 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                             // release.Title += string.Join(" ", titleElements);
                             release.Title = release.Title.Trim();
 
-                            release.Title += " " + titleElements[5] + " " + titleElements[3] + " " + titleElements[1] + " " + titleElements[2] + " " + titleElements[4] + " " + string.Join(" ", titleElements.Skip(6).Take(titleElements.Length - 6).ToArray());
+                            if (titleElements.Length < 6)
+                            {
+                                // Usually non movies / series could have less than 6 elements, eg: Books.
+                                release.Title += " " + string.Join(" ", titleElements.Take(titleElements.Length).ToArray());
+                            }
+                            else
+                            {
+                                release.Title += " " + titleElements[5] + " " + titleElements[3] + " " + titleElements[1] +
+                                                 " " + titleElements[2] + " " + titleElements[4] + " " + string.Join(
+                                                     " ", titleElements.Skip(6).Take(titleElements.Length - 6).ToArray());
```

Comment:
Simiarly here, we're taking the rest of the array after `Skip(6)`, so `Take()` isn't necessary. Also, `string.Join` has an overload that takes an `IEnumerable<string>` type, so there's no need to use `ToArray()`. Not your fault, that was what the original code had. We're trying to clean up things like this because the original code base is a hodge-podge mess. So while we're touching this code, might as well fix it too.
```suggestion
                                                 " " + titleElements[2] + " " + titleElements[4] + " " +
                                                  string.Join(" ", titleElements.Skip(6));
```